# **Tarea Nº 3**
Scope: If, Loops, Functions

# Nueva sección

In [112]:
from google.colab import drive
drive.mount('/content/drive')

MessageError: Error: credential propagation was unsuccessful

Curso:          Fundamentos de Programación en Python para Macroeconomía y Finanzas     
Profesor:       Victor Fuentes Campos     

- Fecha y hora de entrega: Miércoles 10 de enero, 11:59am
- Agregue los nombres de las personas con las que discutió esta tarea: Rojas Chavez Angel Emanuel
- Envíe su tarea haciendo el `push` de su código a su repo en GitHub Classroom:

As of 2022 Q2, there are almost 3.5 million Andriod apps on Google Play and 1.6 million iOS apps on the App Store, and counting. Giving the digital revolution, there is an increase interest in investment opportunities in this sector

<img src="popular_app_stores.png" max-width="300">

Under this context, you will assess a financial committee interested in investing in software development. The goal is to determine the kinds of apps that are likely to attract more users because the number of people using our apps affect our revenue. For this purpose, you will access to data from Google Play and App Store.

- Google Play: you can dowload the dataset containing approximately 11k Android apps from [this link](https://dq-content.s3.amazonaws.com/350/googleplaystore.csv). It was collected in August 2018
- App Store: you can dowload the dataset containing roughly 7k iOS apps from [this link](https://dq-content.s3.amazonaws.com/350/AppleStore.csv). It was collected in July 2017

## Exploring data

In [114]:
import numpy as np
import pandas as pd


1. Open the two datasets we mentioned above, and save both as lists of lists

In [115]:
pip install -r"/content/requirements.txt"

In [116]:
#Solution
google = pd.read_csv("/content/googleplaystore.csv")
apple = pd.read_csv("/content/AppleStore.csv")

2. Print the first few rows of each dataset and find the number of rows and columns of each dataset

In [117]:
# Solution
google.head()


App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [118]:
#Solution
apple.head()

id               track_name  size_bytes currency  price  \
0  284882215                 Facebook   389879808      USD    0.0   
1  389801252                Instagram   113954816      USD    0.0   
2  529479190           Clash of Clans   116476928      USD    0.0   
3  420009108               Temple Run    65921024      USD    0.0   
4  284035177  Pandora - Music & Radio   130242560      USD    0.0   

   rating_count_tot  rating_count_ver  user_rating  user_rating_ver      ver  \
0           2974676               212          3.5              3.5     95.0   
1           2161558              1289          4.5              4.0    10.23   
2           2130805               579          4.5              4.5  9.24.12   
3           1724546              3842          4.5              4.0    1.6.2   
4           1126879              3594          4.0              4.5    8.4.1   

  cont_rating        prime_genre  sup_devices.num  ipadSc_urls.num  lang.num  \
0          4+  Social Networking               37                1        29   
1         12+      Photo & Video               37                0        29   
2          9+              Games               38                5        18   
3          9+              Games               40                5         1   
4         12+              Music               37                4         1   

   vpp_lic  
0        1  
1        1  
2        1  
3        1  
4        1

In [119]:
google.colum = len(google.columns)
google.row= len(apple)
apple.colum = len(google.columns)
apple.row= len(apple)
print(google.colum,google.row,apple.colum,apple.row)

13 7197 13 7197


For more info about the columns in the App Store dataset, visit [this link](https://www.kaggle.com/datasets/ramamet4/app-store-apple-data-set-10k-apps?select=AppleStore.csv)

## Preparing data

3. As noted before in the [discussion forum](https://www.kaggle.com/datasets/lava18/google-play-store-apps/discussion/66015), there is an error in the Google Play dataset. Please, remove the entry 10472 and explain shortly why

In [120]:
# Solution
google.iloc[10472] # Se debe eliminar porque faltan datos

App               Life Made WI-Fi Touchscreen Photo Frame
Category                                              1.9
Rating                                               19.0
Reviews                                              3.0M
Size                                               1,000+
Installs                                             Free
Type                                                    0
Price                                            Everyone
Content Rating                                        NaN
Genres                                  February 11, 2018
Last Updated                                       1.0.19
Current Ver                                    4.0 and up
Android Ver                                           NaN
Name: 10472, dtype: object

In [121]:
google.drop(10472, inplace = True)
google.iloc[10472] #Se logró eliminar la fila


App               osmino Wi-Fi: free WiFi
Category                            TOOLS
Rating                                4.2
Reviews                            134203
Size                                 4.1M
Installs                      10,000,000+
Type                                 Free
Price                                   0
Content Rating                   Everyone
Genres                              Tools
Last Updated               August 7, 2018
Current Ver                       6.06.14
Android Ver                    4.4 and up
Name: 10473, dtype: object

In [122]:
apple.drop(10472, inplace = True)
apple.iloc[10472] #Se logró eliminar la fila

KeyError: '[10472] not found in axis'

4. Count the number of duplicates in the Google Play dataset. You must build a function to loop through the Google Play dataset

In [140]:
# Solution

def contar_duplicados(google):

    total_filas = len(google)

    filas_unicas = len(google.drop_duplicates())
    filas_duplicadas = total_filas - filas_unicas
    return filas_duplicadas

num_duplicados = contar_duplicados(google)
print("Número de filas duplicadas: ", num_duplicados)



Número de filas duplicadas:  483


5. Remove duplicates ...

5. a. Create a dictionary, where each dictionary key is a unique app name and the corresponding dictionary value is the highest number of reviews of that app


In [123]:
# Solution

app_reviews = google.set_index('App')['Reviews'].to_dict()

for app_name, num_reviews in app_reviews.items():
    print(f"Aplicación: {app_name}, Número de reseñas: {num_reviews}")

Se han truncado las últimas 5000 líneas del flujo de salida.
Aplicación: Asylum Night Shift - Five Nights Survival, Número de reseñas: 59223
Aplicación: Five Nights at Freddy's: SL, Número de reseñas: 16162
Aplicación: Two Nights at jumpscare, Número de reseñas: 596
Aplicación: Five Nights at Pizzeria, Número de reseñas: 1976
Aplicación: Block Battles: Heroes at War - Multiplayer PVP, Número de reseñas: 6698
Aplicación: Five Nights at Freddy's 4, Número de reseñas: 21266
Aplicación: Five Nights at Flappy's, Número de reseñas: 4041
Aplicación: 5 Nights at Cube Pizzeria City, Número de reseñas: 2691
Aplicación: Five Nights at Freddy's 3, Número de reseñas: 27856
Aplicación: Undertale AU Amino, Número de reseñas: 5369
Aplicación: au, Número de reseñas: 9126
Aplicación: Audiobooks from Audible, Número de reseñas: 568922
Aplicación: Au Mobile: Audition Chính Hiệu, Número de reseñas: 85278
Aplicación: AU Mobile Indonesia, Número de reseñas: 16521
Aplicación: Au Bon Pain, Número de reseñas: 7

5. b. Use the information stored in the dictionary and create a new dataset, which will have only one entry per app (and for each app, we'll only select the entry with the highest number of reviews).


In [135]:
google_play_unico = pd.DataFrame()

for app_name, max_reviews in app_reviews.items():
    fila = google[(google['App'] == app_name) & (google['Reviews'] == max_reviews)]

    google_play_unico = google_play_unico.append(fila)


Se han truncado las últimas 5000 líneas del flujo de salida.
<ipython-input-135-1b4e97ca32df>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  google_play_unico = google_play_unico.append(fila)
<ipython-input-135-1b4e97ca32df>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  google_play_unico = google_play_unico.append(fila)
<ipython-input-135-1b4e97ca32df>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  google_play_unico = google_play_unico.append(fila)
<ipython-input-135-1b4e97ca32df>:6: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  google_play_unico = google_play_unico.append(fila)
<ipython-input-135-1b4e97ca32df>:6: FutureWarning: 

5. c. Explore the cleaned dataset, it should have 9,659 rows

In [139]:
# Solution
len(google_play_unico.drop_duplicates())


9665

6. Remove non-english apps from both Android and Apple datasets. You should write a function that takes a string, if there's more than three characters that fall outside the ASCII range (0 - 127), then the function should return False (identify the string as non-English), otherwise it should return True. Hint: use the [```ord()```  function](https://www.w3schools.com/python/ref_func_ord.asp)

In [141]:
# For instance
correct_english_name = "Instachat 😜"  # 1 character outside the ASCII range (😜)
incorrect_english_name = "爱奇艺PPS -《欢乐颂2》电视剧热播" # 13 character outside the ASCII range (ie 播)

In [142]:
print(ord("I"))
print(ord("t"))
print(ord("-"))
print(ord("😜"))
print(ord("爱"))

73
116
45
128540
29233


In [144]:
# Solution
def es_ingles(google):
    caracteres_no_ascii = 0

    for caracter in google:
        if ord(caracter) > 127:
            caracteres_no_ascii += 1

    if caracteres_no_ascii > 3:
        return False
    else:
        return True

In [152]:
android_ingles = google[google['App'].apply(es_ingles)]
apple_ingles = apple[apple['track_name'].apply(es_ingles)]
android_ingles

App             Category  \
0         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
1                                    Coloring book moana       ART_AND_DESIGN   
2      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
3                                  Sketch - Draw & Paint       ART_AND_DESIGN   
4                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
...                                                  ...                  ...   
10836                                   Sya9a Maroc - FR               FAMILY   
10837                   Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                             Parkinson Exercices FR              MEDICAL   
10839                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating Reviews                Size     Installs  Type Price  \
0         4.1     159                 19M      10,000+  Free     0   
1         3.9     967                 14M     500,000+  Free     0   
2         4.7   87510                8.7M   5,000,000+  Free     0   
3         4.5  215644                 25M  50,000,000+  Free     0   
4         4.3     967                2.8M     100,000+  Free     0   
...       ...     ...                 ...          ...   ...   ...   
10836     4.5      38                 53M       5,000+  Free     0   
10837     5.0       4                3.6M         100+  Free     0   
10838     NaN       3                9.5M       1,000+  Free     0   
10839     4.5     114  Varies with device       1,000+  Free     0   
10840     4.5  398307                 19M  10,000,000+  Free     0   

      Content Rating                     Genres      Last Updated  \
0           Everyone               Art & Design   January 7, 2018   
1           Everyone  Art & Design;Pretend Play  January 15, 2018   
2           Everyone               Art & Design    August 1, 2018   
3               Teen               Art & Design      June 8, 2018   
4           Everyone    Art & Design;Creativity     June 20, 2018   
...              ...                        ...               ...   
10836       Everyone                  Education     July 25, 2017   
10837       Everyone                  Education      July 6, 2018   
10838       Everyone                    Medical  January 20, 2017   
10839     Mature 17+          Books & Reference  January 19, 2015   
10840       Everyone                  Lifestyle     July 25, 2018   

              Current Ver         Android Ver  
0                   1.0.0        4.0.3 and up  
1                   2.0.0        4.0.3 and up  
2                   1.2.4        4.0.3 and up  
3      Varies with device          4.2 and up  
4                     1.1          4.4 and up  
...                   ...                 ...  
10836                1.48          4.1 and up  
10837                 1.0          4.1 and up  
10838                 1.0          2.2 and up  
10839  Varies with device  Varies with device  
10840  Varies with device  Varies with device  

[10795 rows x 13 columns]

In [153]:
apple_ingles

id                                    track_name  size_bytes  \
0      284882215                                      Facebook   389879808   
1      389801252                                     Instagram   113954816   
2      529479190                                Clash of Clans   116476928   
3      420009108                                    Temple Run    65921024   
4      284035177                       Pandora - Music & Radio   130242560   
...          ...                                           ...         ...   
7182  1070854722                                  Be-be-bears!   480781312   
7186  1169971902                   Hey Duggee: We Love Animals   136347648   
7192  1170406182  Shark Boom - Challenge Friends with your Pet   245415936   
7194  1070052833                                    Go!Go!Cat!    91468800   
7195  1081295232   Suppin Detective: Expose their true visage!    83026944   

     currency  price  rating_count_tot  rating_count_ver  user_rating  \
0         USD   0.00           2974676               212          3.5   
1         USD   0.00           2161558              1289          4.5   
2         USD   0.00           2130805               579          4.5   
3         USD   0.00           1724546              3842          4.5   
4         USD   0.00           1126879              3594          4.0   
...       ...    ...               ...               ...          ...   
7182      USD   2.99                 0                 0          0.0   
7186      USD   2.99                 0                 0          0.0   
7192      USD   0.00                 0                 0          0.0   
7194      USD   0.00                 0                 0          0.0   
7195      USD   0.00                 0                 0          0.0   

      user_rating_ver      ver cont_rating        prime_genre  \
0                 3.5     95.0          4+  Social Networking   
1                 4.0    10.23         12+      Photo & Video   
2                 4.5  9.24.12          9+              Games   
3                 4.0    1.6.2          9+              Games   
4                 4.5    8.4.1         12+              Music   
...               ...      ...         ...                ...   
7182              0.0  1.0.2.5          4+              Games   
7186              0.0      1.2          4+              Games   
7192              0.0    1.0.9          4+              Games   
7194              0.0    1.1.2         12+              Games   
7195              0.0    1.0.3         12+      Entertainment   

      sup_devices.num  ipadSc_urls.num  lang.num  vpp_lic  
0                  37                1        29        1  
1                  37                0        29        1  
2                  38                5        18        1  
3                  40                5         1        1  
4                  37                4         1        1  
...               ...              ...       ...      ...  
7182               35                5        13        1  
7186               40                5         1        1  
7192               38                5         1        1  
7194               37                2         2        1  
7195               40                0         1        1  

[6183 rows x 16 columns]

7. Loop through each dataset to isolate the free apps in separate lists. Make sure you identify the columns describing the app price correctly. Be careful, prices come up as strings ('0', $0.99, $2.99, etc.), so make sure you're not checking an integer or a float in your conditional statements. How many apps are free?


In [169]:

google_gratis = []
apple_gratis = []


for index, row in google.iterrows():
    app_name = row['App']
    price = row['Price']

    if isinstance(price, str):
        price = price.replace('$', '')
        price = float(price)

    if price == 0.0:
        google_gratis.append(app_name)

for index, row in apple.iterrows():
    app_name = row['track_name']
    price = row['price']

    if isinstance(price, str):
        price = price.replace('$', '')
        price = float(price)

    if price == 0.0:
        apple_gratis.append(app_name)

google.drop(27, inplace=True)
google_gratis

['Photo Editor & Candy Camera & Grid & ScrapBook',
 'Coloring book moana',
 'U Launcher Lite – FREE Live Cool Themes, Hide Apps',
 'Sketch - Draw & Paint',
 'Pixel Draw - Number Art Coloring Book',
 'Paper flowers instructions',
 'Smoke Effect Photo Maker - Smoke Editor',
 'Infinite Painter',
 'Garden Coloring Book',
 'Kids Paint Free - Drawing Fun',
 'Text on Photo - Fonteee',
 'Name Art Photo Editor - Focus n Filters',
 'Tattoo Name On My Photo Editor',
 'Mandala Coloring Book',
 '3D Color Pixel by Number - Sandbox Art Coloring',
 'Learn To Draw Kawaii Characters',
 'Photo Designer - Write your name with shapes',
 '350 Diy Room Decor Ideas',
 'FlipaClip - Cartoon animation',
 'ibis Paint X',
 'Logo Maker - Small Business',
 "Boys Photo Editor - Six Pack & Men's Suit",
 'Superheroes Wallpapers | 4K Backgrounds',
 'Mcqueen Coloring pages',
 'HD Mickey Minnie Wallpapers',
 'Harley Quinn wallpapers HD',
 'Colorfit - Drawing & Coloring',
 'Animated Photo Editor',
 'Pencil Sketch Drawing',

In [167]:
apple_gratis

['Facebook',
 'Instagram',
 'Clash of Clans',
 'Temple Run',
 'Pandora - Music & Radio',
 'Pinterest',
 'Bible',
 'Candy Crush Saga',
 'Spotify Music',
 'Angry Birds',
 'Subway Surfers',
 'Solitaire',
 'CSR Racing',
 'Crossy Road - Endless Arcade Hopper',
 'Injustice: Gods Among Us',
 'Hay Day',
 'PAC-MAN',
 'Calorie Counter & Diet Tracker by MyFitnessPal',
 'DragonVale',
 'The Weather Channel: Forecast, Radar & Alerts',
 'Head Soccer',
 'Google – Search made just for mobile',
 'Despicable Me: Minion Rush',
 'The Sims™ FreePlay',
 'Google Earth',
 'Sonic Dash',
 'Groupon - Deals, Coupons & Discount Shopping App',
 '8 Ball Pool™',
 'Tiny Tower - Free City Building',
 'Jetpack Joyride',
 'Bike Race - Top Motorcycle Racing Games',
 'Shazam - Discover music, artists, videos & lyrics',
 'Kim Kardashian: Hollywood',
 'Trivia Crack',
 'WordBrain',
 'Sniper 3D Assassin: Shoot to Kill Gun Game',
 'Flow Free',
 'Lose It! – Weight Loss Program and Calorie Counter',
 'Skype for iPhone',
 'Geometry

In [ ]:
print("Número de aplicaciones gratuitas en Google: ", len(google_gratis))
print("Número de aplicaciones gratuitas en Apple: ", len(apple_gratis))

## Exploring data

 8. Build a frequency table for the ```prime_genre``` column of the App Store dataset, and the ```Genres``` and ```Category``` columns of the Google Play data set. What are the most common genres for each market? You must develop and deploy your function

In [171]:
def tabla_frecuencias(df, columna):

    frecuencias = {}

    for index, row in df.iterrows():
        valor = row[columna]


        if valor in frecuencias:
            frecuencias[valor] += 1
        else:
            frecuencias[valor] = 1

    return frecuencias

frecuencias_apple = tabla_frecuencias(apple, 'prime_genre')
frecuencias_google_genres = tabla_frecuencias(google, 'Genres')
frecuencias_google_category = tabla_frecuencias(google, 'Category')

frecuencias_apple

{'Social Networking': 167,
 'Photo & Video': 349,
 'Games': 3862,
 'Music': 138,
 'Reference': 64,
 'Health & Fitness': 180,
 'Weather': 72,
 'Utilities': 248,
 'Travel': 81,
 'Shopping': 122,
 'News': 75,
 'Navigation': 46,
 'Lifestyle': 144,
 'Entertainment': 535,
 'Food & Drink': 63,
 'Sports': 114,
 'Book': 112,
 'Finance': 104,
 'Education': 453,
 'Productivity': 178,
 'Business': 57,
 'Catalogs': 10,
 'Medical': 23}

In [172]:
frecuencias_google_genres

{'Art & Design': 57,
 'Art & Design;Pretend Play': 2,
 'Art & Design;Creativity': 7,
 'Art & Design;Action & Adventure': 2,
 'Auto & Vehicles': 85,
 'Beauty': 53,
 'Books & Reference': 231,
 'Business': 460,
 'Comics': 59,
 'Comics;Creativity': 1,
 'Communication': 387,
 'Dating': 234,
 'Education;Education': 50,
 'Education': 549,
 'Education;Creativity': 7,
 'Education;Music & Video': 5,
 'Education;Action & Adventure': 6,
 'Education;Pretend Play': 23,
 'Education;Brain Games': 5,
 'Entertainment': 623,
 'Entertainment;Music & Video': 27,
 'Entertainment;Brain Games': 8,
 'Entertainment;Creativity': 3,
 'Events': 64,
 'Finance': 366,
 'Food & Drink': 127,
 'Health & Fitness': 341,
 'House & Home': 88,
 'Libraries & Demo': 85,
 'Lifestyle': 381,
 'Lifestyle;Pretend Play': 1,
 'Adventure;Action & Adventure': 13,
 'Arcade': 220,
 'Casual': 193,
 'Card': 48,
 'Casual;Pretend Play': 31,
 'Action': 365,
 'Strategy': 107,
 'Puzzle': 140,
 'Sports': 398,
 'Music': 22,
 'Word': 29,
 'Racing'

In [173]:
frecuencias_google_category

{'ART_AND_DESIGN': 64,
 'AUTO_AND_VEHICLES': 85,
 'BEAUTY': 53,
 'BOOKS_AND_REFERENCE': 231,
 'BUSINESS': 460,
 'COMICS': 60,
 'COMMUNICATION': 387,
 'DATING': 234,
 'EDUCATION': 156,
 'ENTERTAINMENT': 149,
 'EVENTS': 64,
 'FINANCE': 366,
 'FOOD_AND_DRINK': 127,
 'HEALTH_AND_FITNESS': 341,
 'HOUSE_AND_HOME': 88,
 'LIBRARIES_AND_DEMO': 85,
 'LIFESTYLE': 382,
 'GAME': 1144,
 'FAMILY': 1972,
 'MEDICAL': 463,
 'SOCIAL': 295,
 'SHOPPING': 260,
 'PHOTOGRAPHY': 335,
 'SPORTS': 384,
 'TRAVEL_AND_LOCAL': 258,
 'TOOLS': 843,
 'PERSONALIZATION': 392,
 'PRODUCTIVITY': 424,
 'PARENTING': 60,
 'WEATHER': 82,
 'VIDEO_PLAYERS': 175,
 'NEWS_AND_MAGAZINES': 283,
 'MAPS_AND_NAVIGATION': 137}

In [174]:
genero_mas_comun_apple = max(frecuencias_apple, key=frecuencias_apple.get)
genero_mas_comun_google_genres = max(frecuencias_google_genres, key=frecuencias_google_genres.get)
genero_mas_comun_google_category = max(frecuencias_google_category, key=frecuencias_google_category.get)

print("Género más común en la App Store: ", genero_mas_comun_apple)
print("Género más común en Google Play (Genres): ", genero_mas_comun_google_genres)
print("Género más común en Google Play (Category): ", genero_mas_comun_google_category)

Género más común en la App Store:  Games
Género más común en Google Play (Genres):  Tools
Género más común en Google Play (Category):  FAMILY


 9. One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. Based on that exercise, give one app profile recommendation for Google Play.

    For the Google Play data set, we can find the required information in the ```Installs``` column. Then, compute the average number of installs by genre. Hint 1: consider only unique values of the ```Category``` column. Hint 2: If an app has, for instance, ```1,000,000+``` assume it as ```1,000,000``` installs. Hint 3: Be careful with a skewed distribution

In [176]:
# Solution
instalaciones_por_genero = {}

for index, row in google.iterrows():
    genero = row['Category']
    instalaciones = row['Installs']

    instalaciones = instalaciones.replace('+', '')
    instalaciones = instalaciones.replace(',', '')
    instalaciones = float(instalaciones)

    if genero in instalaciones_por_genero:
        instalaciones_por_genero[genero][0] += instalaciones
        instalaciones_por_genero[genero][1] += 1
    else:
        instalaciones_por_genero[genero] = [instalaciones, 1]

promedio_instalaciones_por_genero = {genero: total_instalaciones / num_apps for genero, (total_instalaciones, num_apps) in instalaciones_por_genero.items()}
promedio_instalaciones_por_genero

{'ART_AND_DESIGN': 1941220.3125,
 'AUTO_AND_VEHICLES': 625061.305882353,
 'BEAUTY': 513151.88679245283,
 'BOOKS_AND_REFERENCE': 8318050.112554112,
 'BUSINESS': 2178075.7934782607,
 'COMICS': 934769.1666666666,
 'COMMUNICATION': 84359886.95348836,
 'DATING': 1129533.3632478632,
 'EDUCATION': 5586230.769230769,
 'ENTERTAINMENT': 19256107.382550336,
 'EVENTS': 249580.640625,
 'FINANCE': 2395215.120218579,
 'FOOD_AND_DRINK': 2156683.0787401577,
 'HEALTH_AND_FITNESS': 4642441.3841642225,
 'HOUSE_AND_HOME': 1917187.0568181819,
 'LIBRARIES_AND_DEMO': 741128.3529411765,
 'LIFESTYLE': 1407443.8193717278,
 'GAME': 30669601.761363637,
 'FAMILY': 5201959.181034483,
 'MEDICAL': 115026.86177105832,
 'SOCIAL': 47694467.46440678,
 'SHOPPING': 12491726.096153846,
 'PHOTOGRAPHY': 30114172.10447761,
 'SPORTS': 4560350.255208333,
 'TRAVEL_AND_LOCAL': 26623593.58914729,
 'TOOLS': 13585731.809015421,
 'PERSONALIZATION': 5932384.647959184,
 'PRODUCTIVITY': 33434177.75707547,
 'PARENTING': 525351.8333333334,


In [177]:
genero_mas_popular = max(promedio_instalaciones_por_genero, key=promedio_instalaciones_por_genero.get)

print("Género más popular en Google Play: ", genero_mas_popular)

Género más popular en Google Play:  COMMUNICATION


 10. The App Store dataset doesn´t have the ```installs``` information. Then, you will construct the average rating from the ```rating_count_tot``` column per genre. Based on that exercise, give one app profile recommendation for the App Store

In [178]:

calificaciones_por_genero = {}


for index, row in apple.iterrows():
    genero = row['prime_genre']
    calificaciones = row['rating_count_tot']


    if genero in calificaciones_por_genero:
        calificaciones_por_genero[genero][0] += calificaciones
        calificaciones_por_genero[genero][1] += 1

    else:
        calificaciones_por_genero[genero] = [calificaciones, 1]

promedio_calificaciones_por_genero = {genero: total_calificaciones / num_apps for genero, (total_calificaciones, num_apps) in calificaciones_por_genero.items()}

genero_mas_popular = max(promedio_calificaciones_por_genero, key=promedio_calificaciones_por_genero.get)

print("Género más popular en la App Store: ", genero_mas_popular)

Género más popular en la App Store:  Social Networking


In [ ]:
# se recomienda la ctegoría Social Networking